cpp/src/arrow/flight/sql/odbc/tests/connection_info_test.cc

```diff
@@ -0,0 +1,865 @@
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoCursorRollbackBehavior) {
+  Validate(this->conn, SQL_CURSOR_ROLLBACK_BEHAVIOR,
+           static_cast<SQLUSMALLINT>(SQL_CB_CLOSE));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoCursorSensitivity) {
+  Validate(this->conn, SQL_CURSOR_SENSITIVITY, static_cast<SQLUINTEGER>(SQL_UNSPECIFIED));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoDataSourceReadOnly) {
+  Validate(this->conn, SQL_DATA_SOURCE_READ_ONLY, static_cast<const SQLWCHAR*>(L"N"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoDefaultTxnIsolation) {
+  Validate(this->conn, SQL_DEFAULT_TXN_ISOLATION, static_cast<SQLUINTEGER>(0));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoDescribeParameter) {
+  Validate(this->conn, SQL_DESCRIBE_PARAMETER, static_cast<const SQLWCHAR*>(L"N"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoMultResultSets) {
+  Validate(this->conn, SQL_MULT_RESULT_SETS, static_cast<const SQLWCHAR*>(L"N"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoMultipleActiveTxn) {
+  Validate(this->conn, SQL_MULTIPLE_ACTIVE_TXN, static_cast<const SQLWCHAR*>(L"N"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoNeedLongDataLen) {
+  Validate(this->conn, SQL_NEED_LONG_DATA_LEN, static_cast<const SQLWCHAR*>(L"N"));
+}
+
+TEST_F(ConnectionInfoMockTest, TestSQLGetInfoNullCollation) {
+  Validate(this->conn, SQL_NULL_COLLATION, static_cast<SQLUSMALLINT>(SQL_NC_START));
+}
+
+TEST_F(ConnectionInfoMockTest, TestSQLGetInfoProcedureTerm) {
+  Validate(this->conn, SQL_PROCEDURE_TERM, static_cast<const SQLWCHAR*>(L""));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoSchemaTerm) {
+  Validate(this->conn, SQL_SCHEMA_TERM, static_cast<const SQLWCHAR*>(L"schema"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoScrollOptions) {
+  Validate(this->conn, SQL_SCROLL_OPTIONS, static_cast<SQLUINTEGER>(SQL_SO_FORWARD_ONLY));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoTableTerm) {
+  Validate(this->conn, SQL_TABLE_TERM, static_cast<const SQLWCHAR*>(L"table"));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoTxnCapable) {
+  Validate(this->conn, SQL_TXN_CAPABLE, static_cast<SQLUSMALLINT>(SQL_TC_NONE));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoTxnIsolationOption) {
+  Validate(this->conn, SQL_TXN_ISOLATION_OPTION, static_cast<SQLUINTEGER>(0));
+}
+
+TEST_F(ConnectionInfoMockTest, TestSQLGetInfoUserName) {
+  Validate(this->conn, SQL_USER_NAME, static_cast<const SQLWCHAR*>(L""));
+}
+
+// Supported SQL
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoAggregateFunctions) {
+  Validate(
+      this->conn, SQL_AGGREGATE_FUNCTIONS,
+      static_cast<SQLUINTEGER>(SQL_AF_ALL | SQL_AF_AVG | SQL_AF_COUNT | SQL_AF_DISTINCT |
+                               SQL_AF_MAX | SQL_AF_MIN | SQL_AF_SUM));
+}
+
+TYPED_TEST(ConnectionInfoTest, TestSQLGetInfoAlterDomain) {
+  Validate(this->conn, SQL_ALTER_DOMAIN, static_cast<SQLUINTEGER>(0));
+}
+
+TYPED_TEST(ConnectionInfoTest, DISABLED_TestSQLGetInfoAlterSchema) {
```

Comment:
Likewise here and for other disabled tests.

Reply:
will remove this test and other disabled tests in future commit